packages/node/src/methods/app-instance/take-action/operation.ts

```diff
@@ -0,0 +1,41 @@
+import { AppInstance } from "@counterfactual/machine";
+import { AppState } from "@counterfactual/types";
+import { Contract } from "ethers";
+import { Provider } from "ethers/providers";
+
+import { ERRORS } from "../../errors";
+
+export async function generateNewAppInstanceState(
+  appInstance: AppInstance,
+  action: any,
+  provider: Provider
+): Promise<AppState> {
+  if (
+    !appInstance.appInterface.addr ||
+    appInstance.appInterface.addr.trim() === ""
+  ) {
```

Comment:
Could we wrap this logic in a util file and call it `isEmptyString`? There may even be an alias on the `Sting` prototype